.cirrus.yml

```diff
@@ -262,3 +243,53 @@ task:
   test_script:
     - ./ci/cirrus.sh
   << : *CAT_LOGS
+
+# Sanitizers
+task:
+  container:
+    dockerfile: ci/linux-debian.Dockerfile
+    cpu: 1
+    memory: 1G
+  env:
+    BUILD:
+    ECDH: yes
+    RECOVERY: yes
+    EXPERIMENTAL: yes
+    SCHNORRSIG: yes
+    CTIMETEST: no
+    EXTRAFLAGS: "--disable-openssl-tests"
+  matrix:
+    - name: "Valgrind (memcheck)"
+      env:
+        # The `--error-exitcode` is required to make the test fail if valgrind found errors, otherwise it'll return 0 (https://www.valgrind.org/docs/manual/manual-core.html)
+        WRAPPER_CMD: "valgrind --error-exitcode=42"
+    - name: "UBSan, ASan, LSan"
+      env:
+        # Use dummy wrapper command to run the tests outside of make check
+        WRAPPER_CMD: env
+        CFLAGS: "-fsanitize=undefined,address"
+        CFLAGS_FOR_BUILD: "-fsanitize=undefined,address"
+        UBSAN_OPTIONS: "print_stacktrace=1:halt_on_error=1"
+        ASAN_OPTIONS: "strict_string_checks=1:detect_stack_use_after_return=1:detect_leaks=1"
+        LSAN_OPTIONS: "use_unaligned=0"
```

Comment:
that's the default according to https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer

Reply:
Hm indeed. To be honest, I can't remember where this is from, it was already in my old branch. I think we should set it to `1` actually. Let me do this.

Some background: Both Valgrind and LSan can check for memory leaks, so we would not need LSan. But LSan is very cheap and none of these tools are perfect. So as long as we don't see false positives, it makes sense to just run LSan even though we have Valgrind. And it turns out that `use_unaligned=1` is indeed one thing where LSan has better coverage than Valgrind: https://stackoverflow.com/questions/33178242/valgrind-and-pragma-pack2  On my machine the difference in running time of `tests 64` was 99,95s with `use_unaligned=0` vs 100,02s `use_unaligned=1`, so this is just noise.

Reply:
cool, sgtm